visualization/app/codeCharta/ui/codeMap/threeViewer/threeSceneService.ts

```diff
@@ -191,18 +190,55 @@ export class ThreeSceneService implements CodeMapPreRenderServiceSubscriber, Map
 
 			hoveredLabel.position.add(this.normedTransformVector)
 
+			this.toggleLineAnimation(false, hoveredLabel)
+
 			this.highlightedLabel = hoveredLabel
 		}
 	}
 
+	resetLineHighlight() {
+		this.highlightedLineIndex = -1
+		this.highlightedLine = null
+	}
+
 	resetLabel() {
 		if (this.highlightedLabel !== null) {
 			this.highlightedLabel.position.sub(this.normedTransformVector)
 			this.highlightedLabel.material.opacity = this.mapLabelColors.alpha
+
+			if (this.highlightedLineIndex > -1) {
+				this.toggleLineAnimation(true)
+			}
+
 			this.highlightedLabel = null
 		}
 	}
 
+	getHoveredLabelLineIndex(labels: Object3D[], label: any) {
+		const index = labels
+			.map(function (labelObject) {
+				return labelObject.uuid
+			})
+			.indexOf(label.uuid)
+
+		if (index >= 0) {
+			return index + 1
```

Comment:
The +1 seems off? Is the index truly starting at 1?

Reply:
Yes, the current implementation puts labels and lines in the same array.
Always label and then corresponding line, so finding a label at index x means that its corresponding label is at x+1.

Reply:
I see. It would likely be nicer to use an array with objects instead but that's just a suggestion.

Reply:
Sorry for the late response, we can change it but it currently affects the whole project, not just this implementation.
So want me to change it?

Reply:
No, let's keep it for now.